execution/gethexec/caff_node.go

```diff
@@ -136,133 +111,45 @@ func (n *CaffNode) createBlock() (returnValue bool) {
 
 	if err != nil {
 		log.Error("Failed to produce block", "err", err)
-		// if we fail to produce a block, we should remove this message from the queue
-		// and set skippedBlockPos to the current messageWithMetadataPos
-		n.skippedBlockPos = &messageWithMetadataPos
-		n.messagesWithMetadata = n.messagesWithMetadata[1:]
-		n.messagesWithMetadataPos = n.messagesWithMetadataPos[1:]
 		return false
 	}
 
 	// If block is nil or receipts is empty, return false
-	if len(receipts) == 0 || block == nil {
+	if block == nil {
 		log.Error("Failed to produce block, no receipts or block")
-		// if we fail to produce a block, we should remove this message from the queue
-		// and set skippedBlockPos to the current messageWithMetadataPos
-		n.skippedBlockPos = &messageWithMetadataPos
-		n.messagesWithMetadata = n.messagesWithMetadata[1:]
-		n.messagesWithMetadataPos = n.messagesWithMetadataPos[1:]
 		return false
 	}
 
-	// Reset the skippedBlockPos
-	n.skippedBlockPos = nil
-
 	blockCalcTime := time.Since(startTime)
 
 	log.Info("Produced block", "block", block.Hash(), "blockNumber", block.Number(), "receipts", len(receipts))
 
 	err = n.executionEngine.appendBlock(block, statedb, receipts, blockCalcTime)
 	if err != nil {
 		log.Error("Failed to append block", "err", err)
+		log.Info("Resetting espresso streamer", "currentMessagePos",
+			messageWithMetadataAndPos.Pos, "currentHostshotBlock",
+			messageWithMetadataAndPos.HotshotHeight)
+		n.espressoStreamer.Reset(messageWithMetadataAndPos.Pos, messageWithMetadataAndPos.HotshotHeight)
```

Comment:
In which scenarios we have to reset the espresso streamer?

In lines above, this function returns `false` without resetting the espresso streamer when failing to produce blocks.

Reply:
We are currently only reseting if we are not able to append to the database. We dont reset if the `ProduceBlock` function returns an error.

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/pull/479/commits/de6b5cdd2f18369b7d59ae36c87307151cf6bc67

I reset the values now

Reply:
https://github.com/EspressoSystems/nitro-espresso-integration/pull/479/commits/de6b5cdd2f18369b7d59ae36c87307151cf6bc67